### PR TITLE
very basic grpc interceptor

### PR DESCRIPTION
**Pull request by dan-compton**
I'm opening this PR to start a conversation around building hatchet GRPC interceptors.  I've quickly thrown this together and you can see the results below.  This implementation is in no way complete or safe to use.

```presence_1  | {"level":"critical","message":"failed to connect to redis","service_name":"presence","service_version":"daff282","time":"2017-11-27T23:02:36.316690172Z"}

presence_1  | {"level":"info","message":"up","service_name":"presence","service_version":"daff282","time":"2017-11-27T23:02:36.317356654Z"}

presence_1  | {"grpc.code":"Unknown","grpc.method":"GetContactAtLocation","grpc.service":"zenreach.nwservices.presence.service.Presence","level":"error","service_name":"presence","service_version":"daff282","time":"2017-11-27T23:02:50.081349626Z"}```

Loosely based on this: https://github.com/grpc-ecosystem/go-grpc-middleware/tree/master/logging (zap)

### CODE REVIEW

File: grpc/interceptors.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	"github.com/zenreach/hatchet"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+// UnaryServerInterceptor returns a new unary server interceptors that logs via a hatchet.Logger.
+func UnaryServerInterceptor(logger hatchet.Logger, opts ...Option) grpc.UnaryServerInterceptor {
+	o := evaluateServerOpt(opts)
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		new := hatchet.Levelize(logger)
```
Comment (BlueDragonX) [inline grpc/interceptors.go]: `new` is a Go keyword so you should name this differently.

File: grpc/options.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
+package grpc_hatchet
+
+import (
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging"
```
Comment (BlueDragonX) [inline grpc/options.go]: Replace this with `grpc_logging "github.com/grpc-ecosystem/go-grpc-middleware/logging"` so it's obvious where the `grpc_logging` package comes from.

File: grpc/options.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
+package grpc_hatchet
```
Comment (BlueDragonX) [inline grpc/options.go]: This should be `grpc`.

File: grpc/interceptors.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package grpc_hatchet
```
Comment (BlueDragonX) [inline grpc/interceptors.go]: This should be `grpc`. Never name the package differently than the directory it is in. That is bad form and confusing. Leave it to the caller to rename the import if they would like.

File: grpc/options.go [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+var (
+	defaultOptions = &options{
+		levelFunc: DefaultCodeToLevel,
+		codeFunc:  grpc_logging.DefaultErrorToCode,
+	}
+)
+
+type options struct {
+	levelFunc CodeToLevel
+	codeFunc  grpc_logging.ErrorToCode
```
Comment (BlueDragonX) [inline grpc/options.go]: No need to suffix these with `Func`. You know they're funcs by their types. There aren't any fields with they might be confused with when lacking the suffix.

**Comment by dan-compton** [comment]
@mrbroll The use case is: I have a service with a grpc interface.  Within that service, I'm already logging some things using a pipelined hatchet.Logger -- assume a broadcastLogger that writes to multiple locations.  This allows you to log all request made to the server via that existing logger.

File: grpc/options.go [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+func WithLevels(f CodeToLevel) Option {
+	return func(o *options) {
+		o.levelFunc = f
+	}
+}
+
+func evaluateServerOpt(opts []Option) *options {
+	optCopy := &options{}
+	*optCopy = *defaultOptions
+	optCopy.levelFunc = DefaultCodeToLevel
```
Comment (BlueDragonX) [inline grpc/options.go]: The effort to copy `defaultOptions` isn't less than simply creating the struct with default values here. Remove `defaultOptions` and just create a new `options` struct with defaults here.

File: grpc/interceptors.go (hunk 1 above)
Comment (dan-compton) [reply grpc/interceptors.go]: whoops..

File: grpc/options.go (hunk 5 above)
Comment (dan-compton) [reply grpc/options.go]: Good point.  I see this from time to time -- same with xChan

File: grpc/interceptors.go [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+	"github.com/zenreach/hatchet"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+// UnaryServerInterceptor returns a new unary server interceptors that logs via a hatchet.Logger.
+func UnaryServerInterceptor(logger hatchet.Logger, opts ...Option) grpc.UnaryServerInterceptor {
+	o := evaluateServerOpt(opts)
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		nl := hatchet.Levelize(logger)
```
Comment (BlueDragonX) [inline grpc/interceptors.go]: You can actually set this outside of the func and clean it up with `logger = hatchet.Levelize(logger)`.

**Pull request merged**